packages/host/app/resources/room.ts

```diff
@@ -4,34 +4,335 @@ import { tracked } from '@glimmer/tracking';
 import { restartableTask } from 'ember-concurrency';
 import { Resource } from 'ember-resources';
 
-import type { RoomField } from 'https://cardstack.com/base/room';
+import { TrackedMap } from 'tracked-built-ins';
 
+import { LooseSingleCardDocument } from '@cardstack/runtime-common';
+
+import { CommandStatus } from 'https://cardstack.com/base/command';
+import type {
+  CardFragmentContent,
+  CardMessageContent,
+  ReactionEvent,
+} from 'https://cardstack.com/base/matrix-event';
+
+import type { MatrixEvent as DiscreteMatrixEvent } from 'https://cardstack.com/base/matrix-event';
+
+import {
+  RoomMember,
+  type RoomMemberInterface,
+} from '../lib/matrix-model/member';
+import { RoomMessageModel } from '../lib/matrix-model/message';
+
+import { RoomModel } from '../lib/matrix-model/room';
+
+import type CardService from '../services/card-service';
 import type MatrixService from '../services/matrix-service';
 
 interface Args {
   named: {
     roomId: string | undefined;
+    events: DiscreteMatrixEvent[] | undefined;
   };
 }
 
+const ErrorMessage: Record<string, string> = {
+  ['M_TOO_LARGE']: 'Message is too large',
+};
+
 export class RoomResource extends Resource<Args> {
-  @tracked room: RoomField | undefined;
+  _messageCache: TrackedMap<string, RoomMessageModel> = new TrackedMap();
```

Comment:
please use `private` modifiers in the resource. otherwise this could be construed as public API (also n oneed to previs with `_` if you say it is `private`